packages/react/counter/src/Counter.tsx

```diff
@@ -39,79 +36,63 @@ export interface CounterProps extends React.ComponentProps<'span'> {
     filters?: FilterInput[]
     /** Propeller that the chart will respond to */
     propeller?: Propeller
+    /** Interval in milliseconds for refetching the data */
+    refetchInterval?: number
+    /** Whether to retry on errors. */
+    retry?: boolean
   }
   /** When true, shows a skeleton loader */
   loading?: boolean
 }
 
 export function Counter(props: CounterProps) {
-  const { value, query, prefixValue, sufixValue, styles, loading = false, localize, ...rest } = props
+  const { value: staticValue, query, prefixValue, sufixValue, styles, loading = false, localize, ...rest } = props
 
   /**
    * If the user passes `value` attribute, it
    * should behave as a static component without any graphql operation performed
    */
   const isStatic = !query
 
-  const [dataValue, setDataValue] = React.useState<string | null>()
   const [propsMismatch, setPropsMismatch] = React.useState(false)
-  const [hasError, setHasError] = React.useState(false)
-  const [isLoading, setIsLoading] = React.useState(false)
 
-  const filtersString = JSON.stringify(query?.filters || [])
   const counterRef = React.useRef<HTMLSpanElement>(null)
 
-  /**
-   * Fetches the counter data
-   * when the user doesn't provide
-   * its own `value`
-   */
-  const fetchData = React.useCallback(async () => {
-    try {
-      setIsLoading(true)
-      setHasError(false)
-
-      const filters = JSON.parse(filtersString)
-
-      const response = await request<CounterQuery, CounterQueryVariables>(
-        PROPEL_GRAPHQL_API_ENDPOINT,
-        CounterDocument,
-        {
-          counterInput: {
-            metricName: query?.metric,
-            timeRange: {
-              relative: query?.timeRange?.relative ?? null,
-              n: query?.timeRange?.n ?? null,
-              start: query?.timeRange?.start ?? null,
-              stop: query?.timeRange?.stop ?? null
-            },
-            filters,
-            propeller: query?.propeller
-          }
-        },
-        {
```

Comment:
As best I can tell, hooks for `dataValue`, `hasError`, `isLoading` are redundant. The `useCounterQuery`, generated by the GraphQL code generator, is already handling this stuff. Additionally, I do not believe we need to stringify the filters, because @tanstack/query handles that for us.

Reply:
The main reason for stringifying the filters was preventing triggering the useEffect in every re-render (due to the filters array reference changing on every re-render) but yes, @tanstack/query handles that for us 😃